app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -24,8 +24,9 @@ def show
         if @record && @record.try(:ems_id)
           ems = ExtManagementSystem.find(@record.ems_id)
           tags = {:type => "node", :hostname => @record.name}.to_json
-          url = polymorphic_path(ems, :display => "ad_hoc_metrics", :tags => tags)
-          redirect_to(url)
+          redirect_to polymorphic_path(ems, :display => "ad_hoc_metrics", :tags => tags)
+        elsif self.class.respond_to?(:custom_display_modes) && self.class.custom_display_modes.index(@display)
+          custom_display_call(@display)
```

Comment:
Agreed, + maybe the whole `ad_hoc_metrics` case should be a show_ad_hoc... method, to keep in line with the others?